src/vmspawn/vmspawn-util.c

```diff
@@ -236,3 +253,35 @@ int find_qemu_binary(char **ret_qemu_binary) {
 
         return find_executable(qemu_arch_specific, ret_qemu_binary);
 }
+
+int machine_cid(unsigned int *ret_machine_cid) {
+        int r;
+        uint64_t cid;
+        _cleanup_close_ int vfd = -EBADF;
+
+        /* Enumerate all machine CIDs in [3..UINT_MAX) until we find a free CID.
+         * if there are no free addresses return SYNTHETIC_ERRNO(EADDRNOTAVAIL)
+         * if ioctl returns an unexpected error forward that
+         */
+
+        r = open("/dev/vhost-vsock", O_RDWR);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to open /dev/vhost-vsock as read/write: %m");
+        vfd = r;
+
+        for (cid = 3; cid < UINT_MAX; cid++) {
+                r = ioctl(vfd, VHOST_VSOCK_SET_GUEST_CID, &cid);
+                if (r >= 0)
+                        goto success;
+                if (errno != EADDRINUSE)
+                        return -errno;
+        }
+
+        return log_debug_errno(SYNTHETIC_ERRNO(EADDRNOTAVAIL), "Enumerated entire CID space and found no free CIDs: %m");
```

Comment:
```suggestion
        return log_debug_errno(SYNTHETIC_ERRNO(EADDRNOTAVAIL), "Enumerated entire CID space and found no free CIDs.");
```

Reply:
The errno is synthetic, so please don't log it.

Reply:
Fixed in 084ed8ddc7.